sheet_to_triples/tests/test_trans.py

```diff
@@ -0,0 +1,225 @@
+        with _mock_open(strdetails) as mock_open, _mock_env() as mock_env:
+            next(trans.Transform.iter_from_name(
+                'test', base_path='/different/path'
+            ))
+
+        mock_open.assert_called_once_with(
+            '/different/path/test.py', 'r', encoding='utf-8'
+        )
+        mock_env.assert_not_called()
+
+    def test_iter_from_name_transform_list(self):
+        transform = ('[{"allow_empty_subject": True},'
+                     '{"allow_empty_subject": False}]')
+
+        with _mock_open(transform), _mock_env():
+            transforms = [t for t in trans.Transform.iter_from_name('test')]
+
+        for t in transforms:
+            self.assertIsInstance(t, trans.Transform)
+
+        self.assertEqual(
+            [t.allow_empty_subject for t in transforms],
+            [True, False]
+        )
+
+    def test_get_non_uniques(self):
+        details = {'non_unique': ['http://a', 'http://b']}
+        transform = trans.Transform('test', details)
+
+        self.assertEqual(
+            transform.get_non_uniques('test-namespace'),
+            {'http://a', 'http://b'}
+        )
+
+    def test_uses_sheet(self):
+        transform = trans.Transform('test', {'sheet': 'test_sheet'})
+        self.assertTrue(transform.uses_sheet())
+
+    def test_uses_sheet_defaults_false(self):
+        self.assertFalse(trans.Transform('test', {}).uses_sheet())
+
+    def test_required_rows(self):
+        details = {
+            'lets': {
+                'testvar': 'prefix_{row[column_1]}',
+                'nonevar': 'no_row_reference',
+            },
+            'triples': [
+                ('{testvar}', 'predicate', '{row[column_2]}')
+            ]
+        }
+        transform = trans.Transform('test', details)
+        self.assertEqual(
+            transform.required_rows(),
+            {'column_1', 'column_2'}
+        )
+
+    def test_required_rows_empty(self):
+        self.assertFalse(trans.Transform('test', {}).required_rows())
+
+    @unittest.skip
+    def test_prepare_queries(self):
+        # unsure how to test this
+        pass
+
+    def test_prepare_queries_no_queries(self):
+        self.assertEqual(
+            trans.Transform('test', {}).prepare_queries('dummy'),
+            {}
+        )
+
+    def _process_and_assertEqual(self, details, expected):
+        rows = trans.Transform('test', details).process({}, details['data'])
+        str_rows = [[str(x) for x in r] for r in rows]
+        self.assertEqual(str_rows, expected)
+
+    def test_process(self):
+        # as we don't care about the internal functioning of rdf in these
+        # tests, we want to just have it return what we give it without
+        # any custom prefix behaviour, which means prepending http://
+        # to subject and predicate values
+        details = {
+            'data': [
+                {'col1': 'http://a', 'col2': '1'},
+                {'col1': 'http://b', 'col2': '2'}
```

Comment:
For actual urls in tests, I prefer sticking to the reserved `.test` TLD which if accidentally resolved won't go anywhere surprising.